syntax = "proto3";

message Clause {
    repeated int32 literal = 1;
}

message Cnf {
    repeated Clause clause = 1;
}

message Interpretation {
    repeated int32 variable = 1;
}

message WhoAreYourParams {}

message ServiceExtended {
    Multihash multihash = 1;
    Service service = 2;
}

message File {
    string file = 1;
}

message Tensor {
}

message Service {

}

service Solver {
    rpc StartTrain ( WhoAreYourParams) returns (WhoAreYourParams) {}
    rpc StopTrain ( WhoAreYourParams) returns (WhoAreYourParams) {}
    rpc GetTensor ( WhoAreYourParams) returns (stream Tensor) {}
    rpc UploadSolver (ServiceExtended) returns (WhoAreYourParams) {}
    rpc StreamLogs (WhoAreYourParams) returns (stream File){}
    rpc Solve(Cnf) returns (Interpretation) {}
}